### PR TITLE
Mapbox TileJSON Probe - for issue #380

### PR DESCRIPTION
**Pull request by jochemthart1**
This PR adds a probe for Mapbox Vector tile service based on issue #380.

The tileset.json is requested and checked for http-errors. 

**Design:**
From the tileset.json, center coordinate and zoom levels are taken to be used in tile requests later on. If center coordinate doesn't exist, the bounds are used to calculate center coordinate. If bounds do not exist either, result is returned as false.

Then, multiple tile url's are constructed using the tile zoom level and center coordinate and are requested and checked for http-errors as well.

An additional check for http204 errors is implemented when requesting tile data, because a request on coordinates with no data would return an empty page.

### CODE REVIEW

**Review by justb4 — CHANGES_REQUESTED** [review]
Thanks @jochemthart1 . Also to @fsteggink : 
My main comment concerns naming/terminology: with "MBTiles" I think of the storage format for both Vector and Raster Tiles in SQLite. Also the [MB docs](https://docs.mapbox.com/help/glossary/mbtiles/#:~:text=MBTiles%20is%20a%20file%20format,from%20many%20different%20data%20sources.) says: _"MBTiles is a file format for storing tilesets. "_. Also MapProxy supports MBTiles file format.

A better naming, also as maybe MB may not be the maintainer in the future (viz MapLibre) **"TileJSON"** is a more adequate name for the web service. As it is also [named TileJSON in the spec](https://github.com/mapbox/tilejson-spec).

I understand this means a lot of renaming but also to avoid later confusion. I propose these name changes:

* `enum`: `MapBox:TileJSON`
* `GeoHealthCheck.plugins.probe.mbtiles.MBTiles` to `GeoHealthCheck.plugins.probe.mapbox.TileJSON`

File: GeoHealthCheck/healthcheck.py [hunk 1]
```diff
@@ -162,6 +162,7 @@ def sniff_test_resource(config, resource_type, url):
                          'FTP': [urlopen],
                          'GHC:Report': [urlopen],
                          'OSGeo:GeoNode': [geonode_get_ows],
+                         'TileJSON': [urlopen],
```
Comment (justb4) [inline GeoHealthCheck/healthcheck.py:165]: Make consistent with (new) definition in `enum.py`: `Mapbox:TileJSON`

File: GeoHealthCheck/enums.py [hunk 2]
```diff
@@ -77,6 +77,9 @@
     'ESRI:FS': {
         'label': 'ESRI ArcGIS FeatureServer (FS)'
     },
+    'MapBox:TileJSON': {
+        'label': 'Mapbox TileJSON Service (TileJSON)'
+    },
```
Comment (justb4) [inline GeoHealthCheck/enums.py:82]: Official company name is `Mapbox` (not `MapBox`), make definition consistent: `Mapbox:TileJSON`.

File: GeoHealthCheck/plugins/probe/mapbox.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+
+class TileJSON(Probe):
+    """
+    TileJSON
+    """
+
+    NAME = 'TileJSON'
+    DESCRIPTION = 'Mapbox TileJSON Service Probe'
+    RESOURCE_TYPE = 'TileJSON'
```
Comment (justb4) [inline GeoHealthCheck/plugins/probe/mapbox.py:12]: Make `RESOURCE_TYPE` consistent: `Mapbox:TileJSON`

File: GeoHealthCheck/config_main.py [hunk 4]
```diff
@@ -161,6 +162,9 @@
     'ESRI:FS': {
         'probe_class': 'GeoHealthCheck.plugins.probe.esrifs.ESRIFSDrilldown'
     },
+    'TileJSON': {
+        'probe_class': 'GeoHealthCheck.plugins.probe.mapbox.TileJSON'
+    },
```
Comment (justb4) [inline GeoHealthCheck/config_main.py:167]: Make consistent: `Mapbox:TileJSON`

**Review by justb4 — CHANGES_REQUESTED** [review]
Main comment: make definition and its use consistent: `Mapbox:TileJSON`.

**Comment by jochemthart1** [comment]
I think automatic checks are failing because of this resource not responding:
[https://geodata.nationaalgeoregister.nl/bag/wms/v1_1](https://geodata.nationaalgeoregister.nl/bag/wms/v1_1)

**Comment by justb4** [comment]
I am trying https://geodata.nationaalgeoregister.nl/beta/topotiles-viewer/styles/tilejson.json 
[on GHC demo](https://demo.geohealthcheck.org/resource/413), but this fails. It advertises v2.2.0. Or is there another root URL to be used?

**Comment by jochemthart1** [comment]
@justb4 I see.
In the design I went with getting the center of the map from the center attribute in the tilejson or from calculating the center from the bounds attribute. However, this endpoint does not seem to have either.

Not sure how I could determine the center tile for this endpoint. Do you have any suggestions?

I might have to add an exception clause there to make sure the error is more clear when this happens.

**Comment by justb4** [comment]
@jochemthart1 I am not so familiar with the TileJSON standard. Found another endpoint from Mapbox, also 2.2.0 with `bounds` but also fails on [GHC demo](https://demo.geohealthcheck.org/resource/414) with another reason: https://a.tiles.mapbox.com/v3/aj.1x1-degrees.json. 
Error `Perform_request Err: UnboundLocalError local variable 'wm_coords' referenced before assignment`.
Not sure what happens. You may consider using `pyproj` (included in GHC) for coordinate  transformation.


`bounds` and `center` are both optional in [the 2.2.0 TileJSON spec](https://github.com/mapbox/tilejson-spec/tree/master/2.2.0). Well, I should say bad luck then, or assume global bounds, but then other failures may arise.

**Pull request merged**